src/actions/menu_items_actions.ts

```diff
@@ -562,3 +562,36 @@ export const CREATE_OR_REMOVE_FILTER_ACTION: ActionSpec = {
     SELECTED_TABLE_HAS_FILTERS(env) ? REMOVE_DATA_FILTER(env) : ADD_DATA_FILTER(env),
   icon: "o-spreadsheet-Icon.FILTER_ICON_ACTIVE",
 };
+
+export function getSendToSheetMenuChildren(
+  env: SpreadsheetChildEnv,
+  sendItemToSheetCallback: (env: SpreadsheetChildEnv, sheetId: UID) => void
+): ActionSpec[] {
+  const sheetIds = env.model.getters
+    .getSheetIds()
+    .filter((sheetId) => sheetId !== env.model.getters.getActiveSheetId());
+
+  const children: ActionSpec[] = [];
+  for (const sheetId of sheetIds) {
+    children.push({
+      name: env.model.getters.getSheetName(sheetId),
+      id: sheetId,
+      icon: "o-spreadsheet-Icon.CLEAR_AND_RELOAD",
+      execute: (env) => sendItemToSheetCallback(env, sheetId),
+    });
+  }
+  children.push({
+    name: _t("New sheet"),
+    id: "new_sheet",
+    icon: "o-spreadsheet-Icon.PLUS",
+    execute: (env) => {
+      const position = env.model.getters.getSheetIds().length;
+      const sheetId = env.model.uuidGenerator.uuidv4();
+      const name = env.model.getters.getNextSheetName(_t("Sheet"));
+
+      env.model.dispatch("CREATE_SHEET", { sheetId, position, name });
+      sendItemToSheetCallback(env, sheetId);
```

Comment:
That makes 2 history steps.
maybe `sheetId` could be optional in commands `SEND_SELECTION_TO_SHEET` and  `SEND_FIGURE_TO_SHEET` and it would create a new sheet in this case?